jstz_api/src/urlpattern/mod.rs

```diff
@@ -0,0 +1,576 @@
+//! `jstz`'s implementation of JavaScript's `URLPattern` Web API.
+//!
+//! More information:
+//!  - [MDN documentation][mdn]
+//!  - [WHATWG `URLPattern` specification][spec]
+//!
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/API/URLPattern
+//! [spec]: https://urlpattern.spec.whatwg.org/
+
+use boa_engine::{
+    js_string,
+    object::{builtins::JsArray, Object},
+    property::Attribute,
+    value::TryFromJs,
+    Context, JsArgs, JsError, JsNativeError, JsObject, JsResult, JsString, JsValue,
+    NativeFunction,
+};
+use boa_gc::{custom_trace, Finalize, GcRefMut, Trace};
+use jstz_core::{
+    accessor,
+    native::{
+        register_global_class, Accessor, ClassBuilder, JsNativeObject,
+        JsNativeObjectToString, NativeClass,
+    },
+    value::IntoJs,
+};
+use urlpattern::quirks::StringOrInit as InnerStringOrQuirksInit;
+use urlpattern::quirks::UrlPatternInit as InnerUrlPatternQuirksInit;
+use urlpattern::UrlPattern as InnerUrlPattern;
+use urlpattern::UrlPatternComponentResult as InnerUrlPatternComponentResult;
+use urlpattern::UrlPatternResult as InnerUrlPatternResult;
+
+pub struct UrlPatternInput(InnerStringOrQuirksInit);
+impl Default for UrlPatternInput {
+    fn default() -> Self {
+        Self(InnerStringOrQuirksInit::Init(
+            InnerUrlPatternQuirksInit::default(),
+        ))
+    }
+}
+#[derive(Default)]
+pub struct UrlPatternInit(InnerUrlPatternQuirksInit);
+
+pub struct UrlPatternComponentResult(InnerUrlPatternComponentResult);
+pub struct UrlPatternResult {
+    // It should be UrlPatternInit instead of UrlPatternInput
+    // according to Deno types?
+    pub(crate) inputs: Vec<UrlPatternInput>,
+    pub(crate) url_pattern_result: InnerUrlPatternResult,
```

Comment:
Nit: Rust naming consistency. Either go for `url_inputs` and `url_pattern_result` or `inputs` and `result`

Reply:
Renamed to `inputs` and `result`.